src/lib/proposer/proposer.ml

```diff
@@ -367,6 +367,11 @@ module Make (Inputs : Inputs_intf) :
                   | `Check_again time ->
                       Singleton_scheduler.schedule scheduler (time_of_ms time)
                         ~f:check_for_proposal
+                  | `Propose_now (time, data) ->
+                      Singleton_supervisor.dispatch proposal_supervisor data
+                      |> ignore ;
+                      Singleton_scheduler.schedule scheduler (time_of_ms time)
```

Comment:
I don't know if it makes sense to handle time like this. It's a little confusing. If you track the most recently checked slot in local state or something, then you don't need to handle this and can just immediately call back into `next_proposal`.

Reply:
How should we prevent a busy loop?